Kernel/Theory.cpp

```diff
@@ -26,307 +26,211 @@
 #include "OperatorType.hpp"
 #include "Term.hpp"
 #include "Kernel/NumTraits.hpp"
+#include "Lib/StringUtils.hpp"
+#include <cstdio>
+#include <iostream>
+
+#if VMINI_GMP
+#include "mini-gmp.c"
+#include "mini-mpq.c"
+#endif
 
 #include "Theory.hpp"
-#define USES_2_COMPLEMENT (~0 == -1)
+
+std::string to_string(mpz_t const& self) {
+  auto s = mpz_sizeinbase(self, /* base */ 10);
+  auto str = new char[s + 2];
+  auto str_ = mpz_get_str(str, /* base */ 10, self);
+  ASS_EQ(str_, str)
+  return std::string(str);
```

Comment:
I think this leaks, as `std::string(char *)` copies the input.

Reply:
Moreover, (as gcc pointed out), `str_` is unused (in release), which makes the code even more suspicious!

Reply:
> Moreover, (as gcc pointed out), `str_` is unused (in Release), which makes the code even more suspicious!

Well the assertion is only very defensive programming, we could actually ignore `str_`. 

> I think this leaks, as std::string(char *) copies the input.

Good poitn! I wasn't aware that that. Fixed now.